napari/components/viewer_model.py

```diff
@@ -914,15 +906,130 @@ def open(
+                # no plugin choice was made
+                else:
+                    layers = self._open_or_raise_error(
+                        [_path], kwargs, layer_type, stack
                     )
+                    added.extend(layers)
+
+        return added
+
+    def _open_or_raise_error(
+        self,
+        paths: List[Path | str],
+        kwargs: Dict[str, Any] = {},
+        layer_type: Optional[str] = None,
+        stack: bool = False,
+    ):
+        """Open paths if plugin choice is unambiguous, raising any errors.
+
+        This function will open paths if there is no plugin choice to be made
+        i.e. there is a preferred reader associated with this file extension,
+        or there is only one plugin available. Any errors that occur during
+        the opening process are raised. If multiple plugins
+        are available to read these paths, an error is raised specifying
+        this.
+
+        Errors are also raised by this function when the given paths are not
+        a list or tuple, or if no plugins are available to read the files.
+        This assumes all files have the same extension, as other cases
+        are not yet supported.
+
+        This function is called from ViewerModel.open, which raises any
+        errors returned. The QtViewer also calls this method but catches
+        exceptions and opens a dialog for users to make a plugin choice.
+
+        Parameters
+        ----------
+        paths : List[Path | str]
+            list of file paths to open
+        kwargs : Dict[str, Any], optional
+            keyword arguments to pass to layer adding method, by default {}
+        layer_type : Optional[str], optional
+            layer type for paths, by default None
+        stack : bool, optional
+            True if files should be opened as a stack, by default False
+
+        Returns
+        -------
+        added
+            list of layers added
+        plugin
+            plugin used to try opening paths, if any
+
+        Raises
+        ------
+        ValueError
+            when paths is *not* a list or tuple, prefered plugin is missing,
+            or no plugins are available
+        RuntimeError
+            when multiple plugins are available or the prefered plugin failed
+        """
+        paths = [os.fspath(path) for path in paths]  # PathObjects -> str
+        if not isinstance(paths, (tuple, list)):
+            raise ValueError(
+                trans._(
+                    "'path' argument must be a string, list, or tuple",
+                    deferred=True,
                 )
+            )
+
+        added = []
+        plugin = None
+        _path = paths[0]
+
+        readers = get_potential_readers(_path)
+        if not readers:
+            raise ValueError(
+                trans._(
+                    'No plugin found capable of reading {paths}.',
+                    deferred=True,
+                    paths=paths,
+                )
+            )
```

Comment:
Since we are in custom-error land already, I'd consider making a NoAvailablePluginForFile error or similar here. Totally not blocking though.